Simplenote/src/androidTest/java/com/automattic/simplenote/integration/TagDialogFragmentTest.kt

```diff
@@ -0,0 +1,326 @@
+package com.automattic.simplenote.integration
+
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.fragment.app.testing.launchFragment
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.RootMatchers.isDialog
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.MediumTest
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import com.automattic.simplenote.BaseUITest
+import com.automattic.simplenote.R
+import com.automattic.simplenote.TagDialogFragment
+import com.automattic.simplenote.utils.TagUtils
+import com.automattic.simplenote.utils.hasTextInputLayoutErrorText
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.not
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+@MediumTest
+class TagDialogFragmentTest : BaseUITest() {
```

Comment:
1) Praise (❤️): Such a nice work well done here, kudos!
2) Suggestion (💡): I wonder since you are checking for the dialog button names to be displayed and asserting their names, maybe we should do the same for the dialog titles as well, like `Tag Conflict` for the `editMergeTagInMultipleNotes` test. PS: The same applies to all tests.
3) Suggestion (💡): I am seeing the formatting of `onView(...)`, `.check(...)`, `.perform(...)`, etc, being done based on the space available in a specific line. I wonder whether formatting these lines to always break on the higher level `dot` might make them more readable. Wdyt? For example:
```
        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name))))
            .check(matches(isDisplayed()))
        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name))))
            .perform(ViewActions.replaceText("tag5"))

        onView(allOf(withText("tag5"), isDescendantOfA(withId(R.id.input_tag_name))))
            .check(matches(isDisplayed()))

        onView(withText(saveText)).inRoot(isDialog())
            .check(matches(isDisplayed()))
            .perform(click())
```

Reply:
> I wonder since you are checking for the dialog button names to be displayed and asserting their names, maybe we should do the same for the dialog titles as well, like Tag Conflict for the editMergeTagInMultipleNotes test. PS: The same applies to all tests.

Fixed in 17d3749d9a22c48cec193209444f7ba6f4ed1a2e.

Reply:
>  I am seeing the formatting of onView(...), .check(...), .perform(...), etc, being done based on the space available in a specific line. I wonder whether formatting these lines to always break on the higher level dot might make them more readable. Wdyt? 


Sounds good. Fixed in a411ab8c18c1d541f97812b4bcc03aa4fed54e17.